t/24-worker-overall.t

```diff
@@ -75,6 +76,15 @@ $ENV{OPENQA_LOGFILE} = undef;
     }
     sub start { }
 }
+{
+    package Test::FakeCacheServiceClientInfo;
+    use Mojo::Base -base;
+    has availability_error => 'Cache service info error: Connection refused';
+}
+
+# mock cache service client
+my $cache_service_client_mock = Test::MockModule->new('OpenQA::CacheService::Client');
+$cache_service_client_mock->redefine(info => sub { Test::FakeCacheServiceClientInfo->new });
```

Comment:
you can also combine these:

```suggestion
Test::MockModule->new('OpenQA::CacheService::Client')->redefine(info => sub { Test::FakeCacheServiceClientInfo->new });
```

Reply:
I can try but I suppose we need to keep at least one reference.

Reply:
I don't think that the instance if the mock module needs to stay preserved. We also have what I suggested in other places, at least within os-autoinst without problems

Reply:
I've just tested it and it doesn't work. You really need to keep the ref.

Reply:
right, I guess

```suggestion
my $cache_service_client_mock = Test::MockModule->new('OpenQA::CacheService::Client')->redefine(info => sub { Test::FakeCacheServiceClientInfo->new });
```

works though.